File: utils/typescript/docs-api-gen/src/markdown-tag.ts
import fs from 'fs';
import path from 'path';
export interface MarkdownTag {
  prefix: string;
  depth: number | 'childOfBefore';
}

// export const Tags: MarkdownTag[] = [H1, H2, H3, H4, H5, UL, UL2, DIVIDER, CUSTOM_CHILD];
const tagsJson = JSON.parse(fs.readFileSync(path.resolve(__dirname, 'res/markdown-tags.json'), 'utf8'));
export const TagsMap: { [key: string]: MarkdownTag } = tagsJson;
console.log(`Markdown Tags: ${TagsMap}`);

export function parseTag(prefix: string, verbose: string): MarkdownTag {
  const ret = Object.values(TagsMap).find((t) => t.prefix === prefix);
  if (!ret) {
    throw new Error(`Unknown tag: ${prefix}, verbose: ${verbose}`);
  }
  return ret;
}

export function generateSubtag(tag: string): string {
  return `<code style={{ position:'relative', top:'-20px', fontSize:'70%'}} children='${tag}'/>`;
}

File: utils/typescript/docs-api-gen/src/python/markdown_tag.py
import json
import os
from typing import Union


class DotDict(dict):
    def __getattr__(self, attr):
        if attr in self:
            return self[attr]
        raise AttributeError(f"'DotDict' object has no attribute '{attr}'")

    def __setattr__(self, attr, value):
        self[attr] = value


dirpath = os.path.dirname(os.path.abspath(__file__))

with open(os.path.join(dirpath, "..", "res/markdown-tags.json"), "r") as file:
    tags_json = json.load(file)

TagsMap = DotDict(tags_json)


def parse_tag(prefix, verbose):
    ret = next((t for t in TagsMap.values() if t["prefix"] == prefix), None)
    if not ret:
        raise ValueError(f"Unknown tag: {prefix}, verbose: {verbose}")
    return ret


def generate_subtag(tag):
    return f"<code style={{{{ position:'relative', top:'-20px', fontSize:'70%'}}}} children='{tag}'/>"
